gh-91118: Fix docstrings that do not honor --without-doc-strings (#31769)
python/cpython: Misc/NEWS.d/next/Documentation/2022-03-08-22-10-38.bpo-46962.FIVe9I.rst

All docstrings in code snippets are now wrapped into :func:`PyDoc_STR` to

> not teach programmers violation of `PEP 7's Documentation Strings paragraph

Reviewer: ```suggestion
follow the guideline of `PEP 7's Documentation Strings paragraph
```
